File: iosApp/iosApp/Services/Settings/SettingsServiceAssembler.swift
// Created by web3d4v on 16/05/2022.
// Copyright (c) 2022 Sons Of Crypto.
// SPDX-License-Identifier: MIT

import web3walletcore

final class SettingsServiceAssembler: AssemblerComponent {
    
    func register(to registry: AssemblerRegistry) {
        
        registry.register(scope: .singleton) { resolver -> SettingsService in
            DefaultSettingsService(
                store: KeyValueStore(name: "\(SettingsService.self)")
            )
        }
    }
}

File: androidApp/src/main/java/com/sonsofcrypto/web3wallet/android/services/SettingsServiceAssembler.kt
package com.sonsofcrypto.web3wallet.android.services

import com.sonsofcrypto.web3lib.keyValueStore.KeyValueStore
import com.sonsofcrypto.web3wallet.android.common.AssemblerComponent
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistry
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistryScope
import com.sonsofcrypto.web3walletcore.services.settings.DefaultSettingsService
import com.sonsofcrypto.web3walletcore.services.settings.SettingsService
import smartadapter.internal.extension.name

class SettingsServiceAssembler: AssemblerComponent {

    override fun register(to: AssemblerRegistry) {

        to.register(SettingsService::class.name, AssemblerRegistryScope.INSTANCE) {
            DefaultSettingsService(
                it.resolve("${KeyValueStore::class.name}.SettingsService"),
            )
        }
    }
}

